### PR TITLE
Test approve

### LINKED ISSUE #216: Test Approve function

### PR DESCRIPTION
**Pull request by Yashk767**
closes #216 
- Added new files : interface.go, structUtils.go, mock.go, approve.go
- Separated approve() function to new file.
- Modified approve to return (txnHash, error)
- Wrote tests for approve function in approve_test.go

### CODE REVIEW

File: cmd/approve_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+					Context:     context.Background(),
+				},
+				transactOpts:    txnOpts,
+				allowanceAmount: big.NewInt(10000),
+				allowanceError:  nil,
+				approveTxn:      &Types.Transaction{},
+				approveError:    nil,
+				hash:            common.BigToHash(big.NewInt(1)),
+			},
+			want: common.Hash{0x00},
```
Comment (ashish10677) [inline cmd/approve_test.go:85]: You missed wantErr here

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: cmd/delegate.go [hunk 2]
```diff
@@ -45,7 +46,12 @@ Example:
 			Config:         config,
 		}
 
-		approve(txnOpts)
+		txnHash, err := approve(txnOpts, razorUtils, tokenManagerUtils, transactionUtils)
+		utils.CheckError("Approve error: ", err)
+		nilHash := common.Hash{0x00}
+		if txnHash != nilHash {
```
[the remark this answers is not in the document]
Comment (Yashk767) [reply cmd/delegate.go:52]: Actually I cannot, if I compare directly it says `Type 'common.Hash' is not an expression`

[the remark this answers is not in the document]
Comment (ashish10677) [reply cmd/delegate.go:52]: As a compilation error or a runtime error?

[the remark this answers is not in the document]
Comment (ashish10677) [reply cmd/delegate.go:52]: If that's the case can you define `nilHash` in core/constants.go and use it everywhere else.

[the remark this answers is not in the document]
Comment (Yashk767) [reply cmd/delegate.go:52]: compilation error

[the remark this answers is not in the document]
Comment (Yashk767) [reply cmd/delegate.go:52]: > If that's the case can you define `nilHash` in core/constants.go and use it everywhere else.

Okay, that works

**Review by ashish10677 — APPROVED** [review]

**Pull request merged**